pandas/core/reshape/melt.py

```diff
@@ -32,7 +33,14 @@ def melt(frame, id_vars=None, value_vars=None, var_name=None,
             raise ValueError('id_vars must be a list of tuples when columns'
                              ' are a MultiIndex')
         else:
+            # Check that `id_vars` are in frame
             id_vars = list(id_vars)
+            missing = Index(id_vars).difference(frame.columns)
+            if not missing.empty:
+                raise KeyError("The following 'id_vars' are not present"
+                               " in the DataFrame:"
+                               " '{missing}'"
+                               "".format(missing=', '.join(missing)))
```

Comment:
This requires the elements of `missing` to be strings I think.

Can you do `.format(missing=list(missing))`?

Reply:
Is `'['missing_A']'` or `['missing_A']` preferred?

Reply:
Doesn't matter too much, but probably the second.